### DISCUSSION TITLE
How to debug Flow Manager application

### DISCUSSION THREAD

**Discussion by rafpor75** [mia-platform/community#495] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

12.0.2

### Description

Hi, 
there is a way to debug locally a Flow Manager application where the microservices are implemented in a Node.js environment?

**Comment by GabrieleGuelfi** [comment]
Hi @rafpor75,

the best way is to look at the logs, they show received events, state transitions and commands triggered, so it is usually enough.
You can find them in the Runtime section of the Console or in an external monitoring tool if you have one (e.g. Grafana).

**Reply by rafpor75** [reply]
my problem is probably something else
run the microservices involved in the flow manager locally and debug them during execution without having to deploy and wait for the pipeline times to speed up development times

**Reply by GabrieleGuelfi** [reply, marked as answer]
You can run a microservice locally and debug it using a "JavaScript Debug Terminal", you can find it in the "Run and Debug" section of Visual Studio Code. If it interacts with the Flow Manager, or any other service, you should mock them using an external tool (e.g. [mock-server](https://github.com/staticdeploy/mock-server) or [Mockoon](https://mockoon.com/)).

**Reply by rafpor75** [reply]
grazie mille :)
